src/goInstallTools.ts

```diff
@@ -506,28 +524,39 @@ function getMissingTools(goVersion: SemVersion): Promise<string[]> {
 // If langserver needs to be used, but is not installed, this will prompt user to install and Reload
 // If langserver needs to be used, and is installed, this will return true
 // Returns false in all other cases
-export function checkLanguageServer(): boolean {
-	let latestGoConfig = vscode.workspace.getConfiguration('go');
-	if (!latestGoConfig['useLanguageServer']) return false;
+export async function checkLanguageServer(): Promise<boolean> {
+	const languageServer = vscode.workspace.getConfiguration('go')['useLanguageServer'];
+	if (languageServer === 'none') return false;
 
 	if (!allFoldersHaveSameGopath()) {
 		vscode.window.showInformationMessage('The Go language server is not supported in a multi root set up with different GOPATHs.');
 		return false;
 	}
 
-	let langServerAvailable = getBinPath('go-langserver') !== 'go-langserver';
+	const langServerAvailable = getBinPath(languageServer) !== languageServer;
 	if (!langServerAvailable) {
-		promptForMissingTool('go-langserver');
+		promptForMissingTool(languageServer);
 		vscode.window.showInformationMessage('Reload VS Code window after installing the Go language server');
+		return false;
 	}
-	return langServerAvailable;
+
+	const activeFileUri = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.document.uri : null;
+	if (activeFileUri !== null) {
+		const isModSupportedResult = await isModSupported(activeFileUri);
+		if (languageServer === 'bingo' && !isModSupportedResult) {
+			vscode.window.showInformationMessage('The Go language server (bingo) does not support non-go module projects. Falling back to default behavior.');
```

Comment:
Check this commit https://github.com/saibing/bingo/commit/f9f872adc38d795840c3c354bd3a6c540f861205
Bingo already supports non-go module projects. I tested it in GOPATH without modules and it was working fine.

Reply:
Thanks for that info @anjmao!
@saibing Can you confirm that non-go module support is ready for use?

Reply:
@anjmao Thanks for the heads up! Kudos to @saibing for adding support for non-go module projects in such a short period of time 😅

@ramya-rao-a I guess I can remove the `isModSupported` checks?

Reply:
@tohjustin Yes, I just tested it out with GOPATH, and it works. So we can remove the `isModSupported` checks.

Reply:
Done